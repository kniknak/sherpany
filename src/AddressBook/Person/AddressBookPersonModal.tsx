import * as React from "react";
import { FetchAddressBookUser } from "../redux/types";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import MuiTypography from "@material-ui/core/Typography";
import withStyles from "@material-ui/core/styles/withStyles";

const Typography = withStyles(theme => ({
    root: {
        margin: theme.spacing(2),
    },
}))(MuiTypography);

export interface AddressBookPersonModalProps {
    person?: FetchAddressBookUser
    isOpen: boolean
    onClose: () => void
}

export class AddressBookPersonModal extends React.PureComponent<AddressBookPersonModalProps> {
    render() {
        const { person, isOpen, onClose } = this.props;

        return (
            <Dialog
                maxWidth="sm"
                fullWidth
                onClose={onClose}
                open={isOpen}
            >
                {person && (
                    <>
                        <DialogTitle>{person.login.username}</DialogTitle>
                        <Typography>location.street.name: {person.location.street.name}</Typography>
                        <Typography>location.street.number: {person.location.street.number}</Typography>
                        <Typography>location.city: {person.location.city}</Typography>
                        <Typography>location.state: {person.location.state}</Typography>
                        <Typography>location.postcode: {person.location.postcode}</Typography>
                        <Typography>phone: {person.phone}</Typography>
                        <Typography>cell: {person.cell}</Typography>
                    </>
                )}
            </Dialog>
        );
    }
}