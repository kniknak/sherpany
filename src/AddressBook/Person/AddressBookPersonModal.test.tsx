import React from "react";
import { shallow } from "enzyme";
import { AddressBookPersonModal, AddressBookPersonModalProps } from "./AddressBookPersonModal";

describe("AddressBook", () => {
    describe("Person", () => {
        describe("AddressBookPersonModal", () => {
            it("renders", () => {
                const props: AddressBookPersonModalProps = {
                    person: {
                        name: {
                            first: "name.first",
                            last: "name.last",
                        },
                        location: {
                            street: {
                                name: "location.street.name",
                                number: 1,
                            },
                            city: "location.city",
                            state: "location.state",
                            postcode: "location.postcode",
                        },
                        email: "email",
                        login: {
                            username: "login.username",
                        },
                        phone: "phone",
                        cell: "cell",
                        id: {
                            value: "id.value",
                        },
                        picture: {
                            thumbnail: "picture.thumbnail",
                        },
                    },
                    isOpen: true,
                    onClose: () => undefined,
                };

                const subject = shallow(<AddressBookPersonModal {...props}/>);

                expect(subject).toMatchSnapshot()
            });
        });
    });
});