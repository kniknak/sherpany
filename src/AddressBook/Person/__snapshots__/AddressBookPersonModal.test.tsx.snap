// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBook Person AddressBookPersonModal renders 1`] = `
<WithStyles(ForwardRef(Dialog))
  fullWidth={true}
  maxWidth="sm"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    login.username
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    location.street.name: 
    location.street.name
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    location.street.number: 
    1
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    location.city: 
    location.city
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    location.state: 
    location.state
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    location.postcode: 
    location.postcode
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    phone: 
    phone
  </WithStyles(WithStyles(ForwardRef(Typography)))>
  <WithStyles(WithStyles(ForwardRef(Typography)))>
    cell: 
    cell
  </WithStyles(WithStyles(ForwardRef(Typography)))>
</WithStyles(ForwardRef(Dialog))>
`;
